packages/router/src/lib/routes.ts

```diff
@@ -0,0 +1,129 @@
+/// <reference types="vite/client" />
+
+import type { Type } from '@angular/core';
+import type { Route } from '@angular/router';
+import { defineRouteMeta } from './define-route';
+
+type Module = {
+  default: Type<unknown>;
+  routeMeta?: ReturnType<typeof defineRouteMeta>;
+};
+
+const FILES = import.meta.glob<Module>(['/app/routes/**/*.ts']);
+
+const ROUTES = Object.keys(FILES).sort((a, b) => a.length - b.length);
+const routeConfigs = ROUTES.reduce<Route[]>((routes: Route[], key: string) => {
+  const module = FILES[key];
+
+  const segments = key
+    .replace(/\/app\/routes|\.(js|ts)$/g, '')
+    .replace(/\[\.{3}.+\]/, '**')
+    .replace(/\[([^\]]+)\]/g, ':$1')
+    .split('/')
+    .filter(Boolean);
+
+  segments.reduce((parent, segment, index) => {
+    const path = segment.replace(/index/g, '').replace('.', '/');
+    const isIndex = !path;
+    const isCatchall = path === '**';
+    const pathMatch = isIndex ? 'full' : undefined;
```

Comment:
suggestion: `pathMatch` default value is `prefix` instead of `undefined`. I think we should keep it as `prefix`

Reply:
Good call